observers/cInputObserver.h

```diff
@@ -0,0 +1,17 @@
+#ifndef D2TM_CINPUTOBSERVER_H
+#define D2TM_CINPUTOBSERVER_H
+
+#include "cMouseObserver.h"
+#include "../include/sKeyboardEvent.h"
+
+class cInputObserver : public cMouseObserver {
+
+public:
+    virtual ~cInputObserver() = default;
+
+    virtual void onNotifyMouseEvent(const s_MouseEvent &event) override = 0;
+    virtual void onNotifyKeyboardEvent(const s_KeyboardEvent &event) = 0;
```

Comment:
This class hierarchy is a little confusing now. I would suggest creating a separate mouse observer and keyboard observer.
For convenience, one could derive an input observer from both, but we could also derive concrete implementations from both. Fortunately, C++ supports multiple inheritance. :)

Reply:
I would prefer a composite over multiple inheritance. But yeah I was lazy and make one depend on the other, while this is not logical to do.